### PR TITLE
web: remove remaining Switch usages

### PR DESCRIPTION
**Pull request by valerybugakov**
## Context

- Migrate remaining `<Switch />` instances to `<Routes />`
- Part of #33834 

## Test plan

1. CI
2. visit affected routes manually

## App preview:

- [Web](https://sg-web-vb-react-router-migration-3.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -0.01% (-0.16 kb) | -0.01% (-0.75 kb) | -0.01% (-0.59 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/vb/react-router-migration-3/compare-report.html;tab=live_object) for a full comparison between the commits 202ddc7 and 9d91d74 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/components/externalServices/ExternalServicesPage.tsx [hunk 1]
```diff
@@ -54,10 +45,9 @@ export const ExternalServicesPage: React.FunctionComponent<React.PropsWithChildr
     })
 
     const editingDisabled = externalServicesFromFile && !allowEditExternalServicesWithFile
-    const isManagingOtherUser = !!userID && userID !== authenticatedUser.id
```
Comment (vovakulikov) [inline client/web/src/components/externalServices/ExternalServicesPage.tsx:57]: Is this logic that we want to remove?

File: client/web/src/components/externalServices/AddExternalServicesPage.tsx [hunk 2]
```diff
@@ -101,13 +90,11 @@ export const AddExternalServicesPage: React.FunctionComponent<
                 <Text>Add repositories from one of these code hosts.</Text>
                 {hasDismissedPrivacyWarning && (
                     <Alert variant="info">
-                        {!userID && (
```
Comment (vovakulikov) [inline client/web/src/components/externalServices/AddExternalServicesPage.tsx:104]: Oh this prop drilling, I couldn't even find this userID prop. Why can we remove this check here? I see in other places you also removed this prop, but I couldn't figure whe we don't need this anymore.

**Review by vovakulikov — APPROVED** [review]
Route changes look good, I checked site admin routes manually, still can't figure out what this userID prop is

File: client/web/src/components/externalServices/AddExternalServicesPage.tsx (hunk 2 above)
Comment (valerybugakov) [reply client/web/src/components/externalServices/AddExternalServicesPage.tsx:104]: I'm in the same boat! I reviewed all the places where the component is used and all the props that we pass to it, and I didn't find any references to `userID`.

Comment (valerybugakov) [reply client/web/src/components/externalServices/AddExternalServicesPage.tsx:104]: Ok, I found it using the diff search. This component was previously reused in user settings routes and `userID` was passed as a prop. Here's [the commit](https://sourcegraph.sourcegraph.com/search?q=context:global+repo:sourcegraph/sourcegraph%24+type:diff+lang:TypeScript+file:client/web/src/user/settings/routes.tsx+AddExternalServicesPage&patternType=standard&sm=1&groupBy=author) that deletes it from there.

Comment (valerybugakov) [reply client/web/src/components/externalServices/AddExternalServicesPage.tsx:104]: Ok, I found it using the diff search. This component was previously reused in user settings routes and `userID` was passed as a prop. Here's [the commit](https://sourcegraph.sourcegraph.com/search?q=context:global+repo:sourcegraph/sourcegraph%24+type:diff+lang:TypeScript+file:client/web/src/user/settings/routes.tsx+AddExternalServicesPage&patternType=standard&sm=1&groupBy=author) that deletes it from there.

File: client/web/src/components/externalServices/ExternalServicesPage.tsx (hunk 1 above)
Comment (valerybugakov) [reply client/web/src/components/externalServices/ExternalServicesPage.tsx:57]: See comment [here](https://github.com/sourcegraph/sourcegraph/pull/47471#discussion_r1102280262).

**Pull request merged**